lib/logic/level.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:hive/hive.dart';
+
+//this class is used to manage the Hive storage box in relation to the statistics page.
+//this class only has class methods.
+class Stats {
+  //a wrapper to set the datafield visitedBoxes of the Hive
+  static void setVisitedArray(Box box, List<dynamic> boxes) {
+    box.put('visitedBoxes', boxes);
+  }
+
+  //a wrapper to get the datafield visitedBoxes of the Hive
+  static List<dynamic> getVisitedArray(Box box) {
+    return box.get('visitedBoxes', defaultValue: []);
+  }
+
+  //a wrapper to add a box to the datafield visitedBoxes of the Hive
+  static void addVisitedBox(Box box, String boxName) {
+    List<dynamic> l = Stats.getVisitedArray(box);
+    l.add(boxName);
+    box.put('visitedBoxes', l);
+  }
+
+  //a wrapper to get the used memory of the chunks stored in the Hive
+  static int getUsedMemory(Box box) {
+    //bytes
+    return box.get('totalSizeInBytes', defaultValue: 0);
+  }
+
+  //a wrapper to get the total usable memory for chunks
+  static double getTotalMemory(Box box) {
+    //megabytes
+    return box.get('dataLimitValue', defaultValue: 10.0);
+  }
+
+  //this function returns the most visited sensor boxes. The first entry in the returned list is the most frequently visited box and so on.
+  static List<dynamic> getTop3Boxes(Box box) {
+    List<dynamic> boxes = []..addAll(Stats.getVisitedArray(box));
+    List<dynamic> top3 = [];
+    top3.add(mostFrequentBox(boxes));
+    boxes = removeFromList(boxes, top3[0]);
+    top3.add(mostFrequentBox(boxes));
+    boxes = removeFromList(boxes, top3[1]);
+
+    top3.add(mostFrequentBox(boxes));
+    boxes = removeFromList(boxes, top3[2]);
+    return top3;
+  }
+
+  //this function returns how many times a given box a has been visited
+  static double getFrequency(String a, List<dynamic> list) {
+    if (a == "1st" || a == "2nd" || a == "3rd") {
+      return 0;
+    }
+    int freq = 0;
+    Map map = Map();
+    list.forEach((x) => map[x] = !map.containsKey(x) ? (1) : (map[x] + 1));
```

Comment:
beschreiben was du hier machst! Warum ist die 1 und das map[x] + 1 nochmal in extra klammern?